src/Microsoft.AspNetCore.Blazor.Server/Builder/BlazorCircuitsApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Blazor.Browser.Rendering;
+using Microsoft.AspNetCore.Blazor.Server.Circuits;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extension methods to configure an <see cref="IApplicationBuilder"/> for Blazor circuits.
+    /// </summary>
+    public static class BlazorCircuitsApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Temporary vague approximation to server-side execution needed so I can
+        /// build the rest of the interop.
+        /// </summary>  
+        public static IApplicationBuilder UseBlazorOnTheServer(
+            this IApplicationBuilder builder,
+            Action<BrowserRenderer> startupAction)
+        {
+            var endpoint = "/_blazor";
+
+            var factory = builder.ApplicationServices.GetRequiredService<CircuitFactory>() as DefaultCircuitFactory;
```

Comment:
Just curious, do you anticipate a scenario where there would be any other type of circuit factory? Is this approach (as opposed to just hard-coding a reference to a particular concrete class here) mainly for consistency with other ASP.NET patterns?

Reply:
Oh wait it looks like you're doing this as a way of supplying a preconfigured object to the SignalR hub. Is that right? If so makes sense.

Reply:
Yes, there are lots of other less coupled ways of doing this as well, but I didn't want to start designing something complicated. `CircuitFactory` is internal anyway and I don't anticipate it becoming a customization point.